pkg/build/controller/build/build_controller.go

```diff
@@ -966,23 +966,54 @@ func (bc *BuildController) resolveImageStreamTag(ref *corev1.ObjectReference, li
+	newRef := ""
+	ok = false
+	if len(stream.Status.DockerImageRepository) > 0 {
+		newRef, ok = imageutil.ResolveLatestTaggedImage(stream, tag)
+	} else {
+		streamCopy := stream.DeepCopy()
+		streamCopy.Status.DockerImageRepository = bc.internalRegistryHostname
+		newRef, ok = imageutil.ResolveLatestTaggedImage(streamCopy, tag)
+	}
+	if ok {
 		// generate informational event if
 		// a) tag has spec
 		// b) tag is local reference
-		// c) status DockerImageRepository is not set, meaning we got this copy of the stream from the watch
+		// c1) status DockerImageRepository is not set, meaning we got this copy of the stream from the watch
 		//    before the image registry was set up
+		// c2) or the import has not happened yet
 		// note, this method is only called for non-output image refs, so we should not get duplicates with
 		// the error/warning currently in place for the output link falling into this category
 		ref2, ok2 := imageutil.SpecHasTag(stream, tag)
-		if ok2 && ref2.ReferencePolicy.Type == imagev1.LocalTagReferencePolicy && len(stream.Status.DockerImageRepository) == 0 {
-			bc.recorder.Eventf(build, corev1.EventTypeNormal,
-				"LocalRefImageStreamTagUnavailable",
-				"The build %s/%s references an image stream tag %s with the local reference policy type, but the internal image registry location was not marked in the image stream",
-				build.Namespace,
-				build.Name,
-				ref.Name)
-		}
+		if ok2 && ref2.ReferencePolicy.Type == imagev1.LocalTagReferencePolicy {
+			if len(stream.Status.DockerImageRepository) == 0 {
+				bc.recorder.Eventf(build, corev1.EventTypeNormal,
+					"LocalRefImageStreamTagUnavailableNoImageRegistryRef",
+					"The build %s/%s references an image stream tag %s with the local reference policy type, but the internal image registry location was not marked in the image stream",
+					build.Namespace,
+					build.Name,
+					ref.Name)
+			}
+			latestTaggedImage := imageutil.LatestTaggedImage(stream, tag)
+			if len(latestTaggedImage.Image) == 0 {
+				//TODO len(latestTaggedImage.Image) == 0 can also result in not using the IST
+				// local tag reference policy;
+				// see https://github.com/openshift/library-go/blob/master/pkg/image/imageutil/helpers.go#L297
+				// see https://github.com/openshift/library-go/blob/master/pkg/image/imageutil/helpers.go#L304
+				// see https://github.com/openshift/library-go/blob/master/pkg/image/imageutil/helpers.go#L333
+				// if stream.Status.DockerImageRepository
+				// is empty, how likely is it that the Image field will be set?
+				// should we do something "crazy" like poll/wait on the stream??
+				// should build controller and image controller have some sort of synchronization,
+				// given they are in the same process, where the build controller puts an event back on the queue
+				// (though for a limited number of retries) if a required imagestream is not ready yet ?
+				bc.recorder.Eventf(build, corev1.EventTypeNormal,
```

Comment:
see above, i don't think we need this, i don't think we get here if the imagestreamtag couldn't be resolved to an imported status.tag, and we already handle that error flow up the chain.

Reply:
responded to your above ... please see and let's reconcile.